.github/workflows/publish_snapshots.yml

```diff
@@ -0,0 +1,53 @@
+# Continuous integration, including test and integration test
+name: Publish snapshots
+
+# Run in master and dev branches and in all pull requests to those branches
+on:
+  push:
+    branches: [ dev ]
+
+jobs:
+  # Build and test the code
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    defaults:
+      run:
+        working-directory: java-sdk
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+
+      - name: Has SNAPSHOT version
+        id: is-snapshot
+        run: grep "version = '.*-SNAPSHOT'" build.gradle
+
+      - uses: actions/setup-java@v1
+        with:
+          java-version: 11
+
+      - name: Cache
+        uses: actions/cache@v2
+        with:
+          # Cache gradle directories
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          # Key for restoring and saving the cache
+          key: ${{ runner.os }}-java-${{ hashFiles('**/*.gradle', 'java-sdk/gradle.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-java-
+
+      - name: Install gpg secret key
+        run: |
+          cat <(echo -e "${{ secrets.OSSRH_GPG_SECRET_KEY }}") | gpg --batch --import
+          gpg --list-secret-keys --keyid-format LONG
+
+      - name: Publish
+        env:
+          OSSRH_USER: ${{ secrets.OSSRH_USER }}
+          OSSRH_PASSWORD: ${{ secrets.OSSRH_PASSWORD }}
+        run: ./gradlew -Psigning.gnupg.keyName=CBEF2CF0 -Psigning.gnupg.executable=gpg -Psigning.gnupg.passphrase=${{ secrets.OSSRH_GPG_SECRET_KEY_PASSWORD }} publish
```

Comment:
usually, there are a lot of commits on dev with the same snapshot versions. Does this ensure that the snapshots with the same version will be updated on maven?

Reply:
Indeed, it will just always send the latest snapshot. In gradle, you can ensure that you have the latest version of a snapshot version by running
```
./gradlew dependencies --refresh-dependencies
```